apps/speed.c

```diff
@@ -175,59 +175,73 @@ typedef struct loopargs_st {
     GCM128_CONTEXT *gcm_ctx;
 } loopargs_t;
 
+typedef int (*loop_proc_fn)(int testnum, loopargs_t *args);
+
+typedef struct loop_ctx_st {
+    int testnum;
```

Comment:
`testnum` was a global var.
It' a counter capped to SIZE_NUM ( count of key-size to test ).
There is a new value of `testnum `passed on each call to each algorithm test-function.
BTW, its use is discard by COND macro, when SIG_ALARM is implemented.

`loop_ctx_st` is a bag to pass two arguments to the async proc.